### ISSUE TITLE
New rule: `no-duplicate-headings-in-section`

### ISSUE THREAD

**Issue by sindresorhus** [wooorm/remark-lint#84]
The following should warn:

``` markdown
## foo

## bar

## foo
```

This should not:

``` markdown
## unicorn

### foo

## rainbow

### foo
```

I don't see any reason not to allow identical headings inside different sections.

**Comment by wooorm** [comment]
The main reason to have this is for linking reasons.

I’ve recently switched to using `h6` headings for `Parameters`, `Returns` for function descriptions in API sections myself, so also sometimes turned them off. But that’s not really perfect either, as it’ll ignore headings on the same level.

Maybe this could also be in [`remark-validate-links`](https://github.com/wooorm/remark-validate-links), that if you refer to, let’s say, `returns`, and there’s a heading in the section with a link pointing to `returns-1`, it’ll warn about it

**Comment by sindresorhus** [comment]
> The main reason to have this is for linking reasons.

Shouldn't that be in the rule description then?

> Maybe this could also be in remark-validate-links, that if you refer to, let’s say, returns, and there’s a heading in the section with a link pointing to returns-1, it’ll warn about it

I don't really see the point of warning about that. Why is `returns-1` wrong? As explained in the issue description having identical headers on different levels makes sense and is often the case.

**Comment by wooorm** [comment]
```md
# foo

## bar

### qux

## baz

### qux
```

Creates the following anchors:

```txt
foo
bar
qux
baz
qux-1
```

So, if someone links to a `qux` heading from the `baz` section like so `[qux](#qux)`, they probably meant to use `#qux-1` instead.

Not sure if that’s always the case, plus it’s a bit unrelated to the issue at hand, but it may solve the problem this rule is trying to solve, better.

Re this particular rule, I think both should be possible, probably with the default staying like this. Possible with a new rule, `no-duplicate-headings-per-section`, having the behaviour you described above.

**Comment by sindresorhus** [comment]
My point is that:

```markdown
## unicorn

### foo

## rainbow

### foo
```

Also result in `-1`:

```
unicorn
foo
rainbow
foo-1
```

**Comment by sindresorhus** [comment]
> Possible with a new rule, no-duplicate-headings-per-section, having the behaviour you described above.

👍
